### PR TITLE
Support fine-grained SLIS filtering during onboarding

### PR DESCRIPTION
**Pull request by emmakarova**
**Description**
Specify additional properties in slis filter label for application template, by which the system fetching for tenants will be fine-grained.

Changes proposed in this pull request:
- introduce slisFilter label and make systemRole required when slisFilter is defined
- when creating application template if systemRole label set, but slisFilter label is missing, create a one that contains the productId and managedBy property with default value
- introduce additional validations for labels in GraphQL mutation for onboarding
- replace getBySystemRoleAndRegion logic with matching by slis filter in EnhanceWithTemplateID in system fetcher
- unit tests
- adjusted e2e tests
- update dependencies

**Related issue(s)**

- ...

**Pull Request status**

- [ ] Implementation
- [ ] Unit tests
- [ ] Integration tests
- [ ] `chart/compass/values.yaml` is updated 
- [ ] Mocks are regenerated, using the automated script

### CODE REVIEW

File: components/director/internal/domain/apptemplate/resolver.go [hunk 1]
```diff
@@ -43,8 +43,17 @@ const (
...
 	displayNameLabelKey        = "displayName"
+	slisFilterLabelKey         = "slisFilter"
 )
 
+var defaultSlisFilterValueForManagedByProperty = []map[string]interface{}{
+	{
+		"key":       "$.additionalAttributes.managedBy",
+		"value":     []string{"SAP Cloud"},
+		"operation": "exclude",
+	},
```
[the remark this answers is not in the document]
Comment (emmakarova) [reply components/director/internal/domain/apptemplate/resolver.go:54]: The value is added as part of the Labels and that is in format map[string]interface. I think the new struct cannot be used in this case

File: components/director/pkg/graphql/app_template_validation.go [hunk 2]
```diff
@@ -13,6 +14,12 @@ import (
 	"github.com/pkg/errors"
 )
 
+const (
+	systemRoleLabelKey = "systemRole"
```
[the remark this answers is not in the document]
Comment (emmakarova) [reply components/director/pkg/graphql/app_template_validation.go]: The validation for the labels input is now moved to the resolver in order to use the env var for systemRole from director deployment

**Review by kaloyanspiridonov — APPROVED** [review]

**Review by PetarTodorovv — APPROVED** [review]
Formal approve, due to go.mod changes only

**Pull request merged**